signer/src/transaction_coordinator.rs

```diff
@@ -2606,15 +2615,28 @@ pub async fn should_coordinate_dkg(
 
 /// Assert, given the last dkg and smart contract current aggregate key, if we
 /// need to verify the shares and/or issue a rotate key call.
-pub fn assert_rotate_key_action(
+pub fn assert_rotate_key_action<C>(
+    context: &C,
     last_dkg: &model::EncryptedDkgShares,
     current_aggregate_key: Option<PublicKey>,
-) -> Result<(bool, bool), Error> {
+    bitcoin_chain_tip: &model::BitcoinBlockRef,
+) -> Result<(bool, bool), Error>
+where
+    C: Context,
+{
     let needs_rotate_key = Some(last_dkg.aggregate_key) != current_aggregate_key;
 
+    // Check if past verification window, if we are, skip verification
+    let dkg_verification_window = context.config().signer.dkg_verification_window;
+    let max_verification_height = last_dkg
+        .started_at_bitcoin_block_height
+        .saturating_add(dkg_verification_window as u64);
+
+    let past_verification_window = bitcoin_chain_tip.block_height > max_verification_height;
+
     let needs_verification = match last_dkg.dkg_shares_status {
-        model::DkgSharesStatus::Unverified => true,
-        model::DkgSharesStatus::Verified => needs_rotate_key,
+        model::DkgSharesStatus::Unverified => !past_verification_window,
+        model::DkgSharesStatus::Verified => needs_rotate_key && !past_verification_window,
```

Comment:
I think we should do something similar for the `needs_rotate_key` logic, where we say it's false if the shares are `Unverified` and we are past the max verification height.